### PR TITLE
[Hmk/url encryption] Fixed static file serving issues

### PR DESCRIPTION
**Pull request by huzaifamk**
## PR Description
Fixed the issue of profile picture not appearing and static file was not being served properly.

## Manual Test Suite
SECTION 1: WEVE GOT POEMS (http://localhost:5173/)
[ ] Log in with 'tester' / '1234'
[ ] Log in anonymously with Layer8
[ ] Clicking "Get Next Poem" loads different poem correctly x 3
[ ] Clicking "Logout" takes you to the login screen
[ ] Clicking "Register" takes you to the registration page
[ ] Registering with a username, password, and profile image is successful
[ ] Logging in with the new username / password succeeds
[ ] Logging in with Layer8 opens the pop-up
[ ] Logging in with the "tester" & "12341234" works
[ ] User chooses to share their new "Username" & "Country" from the Layer8 Resource Server
[ ] Clicking "Get Next Poem" loads different poem correctly x 3
[ ] Clicking "Logout" takes you to the login page

SECTION 2: IMSHARER (http://localhost:5174/)
[ ] Main page loads
[ ] Upload of image works
[ ] Reload leads to instant reload (demonstrating proper caching)
[ ] Clicking the newly loaded image shows it in a light boxLog in using tester and 12341234 should succeed

### CODE REVIEW

File: .github/pull_request_template.md
Comment (DeeStarks) [inline .github/pull_request_template.md:1]: Should we have this here? I thought it was only for use in the layer8 repo?

File: interceptor.go [hunk 1]
```diff
@@ -631,10 +635,33 @@ func getStatic(this js.Value, args []js.Value) interface{} {
...
+		bodyMap := map[string]interface{}{
+			"__url_path": urlPath,
+		}
+
+		bodyByte, err := json.Marshal(bodyMap)
+		if err != nil {
+			reject.Invoke(js.Global().Get("Error").New(err.Error()))
+			return nil
+		}
+
```
Comment (DeeStarks) [inline interceptor.go:659]: Do you mind explaining this process and what's expected? I would like to understand how the added `/media` works, what it changes, etc..

File: internals/main.go [hunk 2]
```diff
@@ -158,7 +158,18 @@ func (c *Client) do(
 	}
 	// create request
 	client := &http.Client{}
-	r, err := http.NewRequest("POST", c.proxyURL+parsedURL.Path, bytes.NewBuffer(data))
+  
+	// !NOTE: GET Requests are also converted to POST requests
+	// fmt.Println("[Interceptor] c.proxyURL+parsedURL.Path: ", c.proxyURL+parsedURL.Path+"?"+parsedURL.RawQuery) // For debugging purposes
+	
+  var url string
```
Comment (DeeStarks) [inline internals/main.go:165]: Could you run `go fmt ./...` to format the code?

```suggestion
    var url string
```

**Review by DeeStarks — CHANGES_REQUESTED** [review]

File: .github/pull_request_template.md
Comment (huzaifamk) [reply .github/pull_request_template.md:1]: I will remove this.

File: interceptor.go (hunk 1 above)
Comment (huzaifamk) [reply interceptor.go:659]: `/media` is for static files, we define the same in the node backend, if you see this code:

`Go
app.use('/media', layer8_middleware.static('pictures'));
`

I am going to test and do some changes for making this dynamic, as I discussed this with Ravi, you will see a commit for it soon!

**Review by DeeStarks — COMMENTED** [review]
The code looks good to me, but I'm still not able to get the media to load. Perhaps I'm not doing something right?

image

image

image

We can hop on a call sometime, maybe

**Comment by huzaifamk** [comment]
> The code looks good to me, but I'm still not able to get the media to load. Perhaps I'm not doing something right?
> 
> image
> 
> image
> 
> image
> 
> We can hop on a call sometime, maybe

Let me test and fix this, let's see what's the issue here.

**Comment by huzaifamk** [comment]
@DeeStarks I have pushed fixes for both middleware and interceptor, and also a new branch is here in layer8, which you can use with the new version of interceptor/middleware: https://github.com/globe-and-citizen/layer8/tree/hmk/url-encrypt-decrypt-feat

**Review by DeeStarks — APPROVED** [review]

**Pull request merged**